From 8494f42a1fae822128ba60dedf826f340cbaf0ae Mon Sep 17 00:00:00 2001
From: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
Date: Sun, 5 May 2024 23:20:01 +0800
Subject: [PATCH 3/3] driver/mmc: add delay during read operation

This is a workaround patch to fix the read operation which
require some delay when using PIO read transfer mode.

Signed-off-by: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
---
 drivers/mmc/efx_sdhost.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/efx_sdhost.c b/drivers/mmc/efx_sdhost.c
index 6ed045e6a3..b0e0d02c02 100644
--- a/drivers/mmc/efx_sdhost.c
+++ b/drivers/mmc/efx_sdhost.c
@@ -212,6 +212,7 @@ static void efx_sdhost_read_block_pio(struct efx_sd_host *host)
 		while ((readl(host->ioaddr + EFX_SDHCI_PRESENT) & EFX_SDHCI_DATA_AVAILABLE) != 0x800);
 
 		for (int i = 0; i < (blksize / 4); i++) {
+			udelay(10);
 			value = readl(host->ioaddr + 0x120);
 
 			/* ACMD51 is a special case */
-- 
2.17.1

